sql/src/main/java/io/crate/analyze/RestoreSnapshotAnalyzer.java

```diff
@@ -120,10 +123,41 @@ public RestoreSnapshotAnalyzedStatement analyze(RestoreSnapshot node, Analysis a
             return RestoreSnapshotAnalyzedStatement.forTables(
                 nameParts.get(1),
                 repositoryName,
-                builder.build(),
+                settings,
                 ImmutableList.copyOf(restoreIndices));
         } else {
-            return RestoreSnapshotAnalyzedStatement.all(nameParts.get(1), repositoryName, builder.build());
+            return RestoreSnapshotAnalyzedStatement.all(nameParts.get(1), repositoryName, settings);
         }
     }
+
+    @VisibleForTesting
+    public static Collection<String> indexNames(TableIdent tableIdent, List<Snapshot> snapshots, boolean ignoreUnavailable) {
+        // If ignoreUnavailable is true, it's cheaper to simply return indexName and the partitioned wildcard instead
+        // checking if it's a partitioned table or not
+        String indexName = tableIdent.indexName();
+        if (ignoreUnavailable) {
+            return Arrays.asList(
+                indexName,
+                PartitionName.templateName(tableIdent.schema(), tableIdent.name()) + "*"
```

Comment:
yes, that works as PartitionName.templateName(...) returns `.partitioned.my_parted.` and so the resulting wildcard contains a dot in the end, like `.partitioned.my_parted.*`